Assignment2

ErikEremenko opened the pull request.

ErikEremenko committed:
Create cmake-single-platform.yml

Used for the CI exercise

ErikEremenko committed:
Added Assignment2 branch to gh action cmake

ErikEremenko committed:
Test CI run

ErikEremenko committed:
Added Doxygen install in github action and reverted files

ErikEremenko committed:
Test new CI action config

ErikEremenko committed:
Test new CI action config with installed vtk and concrete path

ErikEremenko committed:
Github actions CI pipeline with caching + first Google tests (#16)

* CI test with gtest and new CMake configs

* Update CI action to include CI-gtest branch

* Update CI action to include CI-gtest branch, bfix

* Try out caching for github cmake action

* Update repo version for ccache

* Small improvements in testing and Cmake, build script

* Bfix: Make doxygen installation optional when building the project

* Bfix: Make doxygen installation mandatory if actually creating documentation

* Added dependency for doxygen to the github action

* Improve caching in the cmake action

* Fix poisoned cahce by invalidating old cache key in action

* Test cmake action's caching

* Test cmake action with complete vtk import

* Test cmake action with additional vtk import

* Removed hard-coded vtk path from action

* Removed libvtk9.1 from apt install

* Test new caching config

* Invalidate cache and add VTK runtime libraries

* Test new caching config

* Disable build directory caching in cmake action

* Test new caching config

* Invalidate cahce, remove apt installations

* Test new caching config

* Remove vtk from action

* Restore vtk and deps

* Test new action config

* Add caching to action

* Test caching

ErikEremenko committed:
Implemented Lennard Jones Force calculation

ErikEremenko committed:
Added .clang-tidy and implemented some suggestions from clang.

# Conflicts:
#	src/Simulation.h

ErikEremenko committed:
Fixed bugs stemming from the rebase

ErikEremenko committed:
removed nonexistent branch from cmake github action

Commit:
added FileCuboidReaderClass for reading cuboids and a temporary particle generator code segment and added example cuboid input file

Commit:
added FileCuboidReader class with temporary particle generation code and
added example cuboid input file

ErikEremenko committed:
Test new caching in cmake action

ErikEremenko committed:
Bugfix in LJ-Force calculation. Set up for cuboid simulation.

dpet9941 committed:
Fixed temperature calculation in FileCuboidReader

ErikEremenko committed:
Added more test cases and modified ParticleContainer to prevent passing Particle by copy. Minor fixes and refinements for the rest.

ErikEremenko committed:
Merge branch 'Assignment2' of github.com:ErikEremenko/MolSim-WS25-GroupB into Assignment2

berkerdemoglu committed:
Refactor CalcMethod into separate force calculations such as gravity and Lennard-Jones

berkerdemoglu committed:
Move Simulation class method definitions to a separate source file

berkerdemoglu committed:
Add benchmark and file output modes to Simulation

berkerdemoglu committed:
Create include directory and move .h headers there

berkerdemoglu committed:
Rename outputWriter to io and move file reader classes there. Also fix include path name errors in the moved files

berkerdemoglu committed:
Fix refactored name issues in force calculation tests

berkerdemoglu committed:
Add BaseSimulation and BaseFileReader classes

ErikEremenko committed:
Adapted the tests to the new ForceCalc class. Introduced automatic clang formatting to the CI-pipeline.

ErikEremenko committed:
Merge changes

ErikEremenko committed:
style: Apply clang-format

ErikEremenko committed:
Increase Clang-format column limit to 120.

ErikEremenko committed:
style: Apply clang-format

ErikEremenko committed:
Implemented command line passing of the simulation mode

ErikEremenko committed:
Added spdlog to project and refactored the CMake files

ErikEremenko committed:
Fixed benchmark

ErikEremenko committed:
style: Apply clang-format

ErikEremenko committed:
Commented the cuboid input file format

berkerdemoglu committed:
Add cutoff radius to Lennard-Jones potential

berkerdemoglu committed:
Merge remote-tracking branch 'refs/remotes/origin/Assignment2' into Assignment2

berkerdemoglu committed:
Replace console statements with spdlog log statements

ErikEremenko committed:
Improved benchmark script and fixed ForceCalcTest

ErikEremenko committed:
style: Apply clang-format

berkerdemoglu committed:
Add Doxygen comments for ForceCalc, Simulation and FileReader classes

berkerdemoglu committed:
Merge remote-tracking branch 'refs/remotes/origin/Assignment2' into Assignment2

Commit:
added another command line option for defining the log level

ErikEremenko committed:
Implemented LJ-F Parallelization with OpenMP

dpet9941 committed:
style: Apply clang-format

ErikEremenko committed:
Included Doxygen documentatio + LennardJonesForceParallel documentation

ErikEremenko committed:
Merge branch 'Assignment2' of github.com:ErikEremenko/MolSim-WS25-GroupB into Assignment2 to include parallelization

ErikEremenko committed:
style: Apply clang-format

ErikEremenko committed:
Update scripts and set up loggin in main

ErikEremenko committed:
Merge branch 'Assignment2' of github.com:ErikEremenko/MolSim-WS25-GroupB into Assignment2 to add logging functionality in main

Commit:
changed logs in FileReader to spdlogs

Commit:
merge witha logs in FileReader

dpet9941 committed:
style: Apply clang-format

ErikEremenko committed:
Fix logging, Implemented parallelization CLI selection

ErikEremenko committed:
style: Apply clang-format

ErikEremenko committed:
Improved benchmark scripts with performance options and fixed logging

ErikEremenko committed:
Merge branch 'Assignment2' of github.com:ErikEremenko/MolSim-WS25-GroupB into Assignment2 (long +g + chmark scripts)

ErikEremenko committed:
Fixed Doxygen generation

dpet9941 committed:
fixed hashtag lines logs and added particle instatiation logs

ErikEremenko committed:
style: Apply clang-format

ErikEremenko committed:
Fixed bug in multithreaded LJ

ErikEremenko committed:
Minor benchmark fix

ErikEremenko committed:
Update Readme, remove comment

ErikEremenko committed:
Update Readme, fix error log

ErikEremenko committed:
Merge branch 'Assignment2' of github.com:ErikEremenko/MolSim-WS25-GroupB into Assignment2

dpet9941 committed:
added more logs

dpet9941 committed:
Merge remote-tracking branch 'origin/Assignment2' into Assignment2_recent

dpet9941 committed:
style: Apply clang-format

schuhmaj reviewed (approved):
There is not much to comment here. This is perfect.

I think the only thing one can complain: Missing test case for the cuboid generator. Otherwise everything done very well

ErikEremenko committed:
Implemented PR-review suggestions: CuboidFileReader tests, compilation using  cmake --build, s
et release mode to Release in CI-Pipeline, renamed FileCuboidReader to CuboidFileReader for unifromity with BaseFileReader, added project source directory macro in CMa
ke

ErikEremenko committed:
Merge branch 'Assignment2' of github.com:ErikEremenko/MolSim-WS25-GroupB into Assignment2 to incorporate PR-Review changes.

ErikEremenko committed:
style: Apply clang-format
